### ISSUE TITLE
[Bug]: no LogIn with PlugIn possible

### ISSUE THREAD

**Issue by Vollspacken** [homebridge-eufy-security/plugin#201]
### What happened?

My configured Eufy Cam, which previously worked without restriction in HomeKit, was suddenly displayed there as "unavailable".

After various troubleshooting without success I have logged out of the Homebridge PlugIn or reset. 

Now I wanted to log in again with the Eufy guest account to reconfigure the plugin, but I always get the error message: "Login not successful". Also the attempt to log in with the Eufy Admin account fails. However, both Eufy accounts are active, a login on the Eufy Security Portal or in the Eufy App works fine. 

Has something been blocked on the part of Eufy so that a login via Homebridge plugin is no longer possible?

### Device Type

Indoor Camera (Supported)

### Plugin Version

v2.0.x (Supported)

### HomeBridge Version

v1.4.x (Supported)

### NodeJS Version

v16 (Supported)

### Operating System

Raspbian 10+ (Supported)

### Relevant log output

```shell
no log Output
```

### config

```shell
no Config
```

**Comment by r33ce15** [comment]
I have just seen the same issue! I see in the logs just 

ERROR: not connected can't continue 

Not too helpful and yes when trying to re login it doesn't work so something has happened

**Comment by r33ce15** [comment]
> It looks like the API might've changed, but I'm not sure.
> 
> 
> 
> ```
> 
> {"name":"[EufySecurity-2.2.0-beta.5]","hostname":"homebridge","pid":9794,"level":20,"msg":"Request: {\n  method: 'post',\n  endpoint: 'v2/passport/login_sec',\n  token: null,\n  data: {\n    ab: 'US',\n    client_secret_info: {\n      public_key: '04c1e9f6581d5441ae77c11de936d4249fb1a57f0fe2990ed8e3a90d4edf6136308db792c871953da31cf17f2b4e6308c358028086c0eeb2975e38e23291110abd'\n    },\n    enc: 0,\n    email: '<redacted>',\n    password: '<redacted>',\n    time_zone: 0,\n    transaction: '1668022538048'\n  }\n}","time":"2022-11-09T19:35:38.050Z","v":0}
> 
> {"name":"[EufySecurity-2.2.0-beta.5]","hostname":"homebridge","pid":9794,"level":20,"msg":"Response: { response: { code: 0, msg: 'Succeed.' } }","time":"2022-11-09T19:35:38.392Z","v":0}
> 
> {"name":"[EufySecurity-2.2.0-beta.5]","hostname":"homebridge","pid":9794,"level":50,"msg":"Generic Error: TypeError: Cannot read properties of undefined (reading 'user_id')\n    at HTTPApi.login (/var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/http/api.js:252:66)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async EufySecurity.connect (/var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/eufysecurity.js:605:9)","time":"2022-11-09T19:35:38.396Z","v":0}
> 
> ```

I think your right! I also see in bropat's library in which this plugin relies on at beginning on November there was a new build which in change logs states about a login api changed to new api so guess eufy have killed old on so will have to wait for a new build with this dependency updated

**Comment by DolmensDude** [comment]
Happened here too. Motion sensors throughout house no longer triggering any HomeKit automation. That’s a lot!!!

**Comment by andypburke** [comment]
Some has happened to the eufy-security-client as shown here for Home Assistant https://github.com/bropat/eufy-security-client/issues/242.    I think the same issue is happening here with Homebridge, but not sure how to fix it.   It seems to be something to do with CAPTCHA and trusted device variable,

**Comment by csoprano** [comment]
Hello everybody,
I have the same problem too. I can no longer control the alarm system, login error. please solve the problem.
Thanks

**Comment by cracyfloyd** [comment]
The same here, it is very new thats the login is not possible. A few days ago it works well. Nothing was changed by me. Untouched System.

**Comment by msosio** [comment]
I have the same issue. Can't login anymore.

**Comment by DolmensDude** [comment]
clearly, there is an issue - I've used the eufy security app for Homey on that system as a stopgap for some of the sensors, but this plugin is far superior to that one which is why I use it.

Hopefully folks can get on this...  and of course, if eufy is actually blocking this method of client access, or more importantly for this particular client app, then it is incumbent on eufy to actually implement HomeKit on products they have been saying for a year+ they'd be supporting!!!

**Comment by zsozsobasa** [comment]
Yes, this issue is for everybody momently. They will fix - hope so...

**Comment by TheAdrianMc** [comment]
Ok I'm glad I wasn't going crazy on why my camera just stopped working! I'm sure a fix is on the way. Luckily I don't rely on having my cam on HK.

**Comment by Sylanorr** [comment]
Same issue here since i update homebridge and reset the instance.

**Comment by callumcook2000** [comment]
same issue here would worked flawlessly up until yesterday, seems to be on eufy's end

**Comment by Mech4711** [comment]
Can we somehow edit some files of the installed plugin to get that change working?
I rely on having all my cameras accessible in HomeKit and don't want to wait for the pull request to get approved

**Comment by Lokihb34** [comment]
Yeah, I have a Eufy doorbell in Homebridge and I simply performed an update to my software 2 days ago, same time this thread open, and I tried reinstalling the plugin and Homebridge on 3 different devices, but get login error when trying to sign into eufy plugin... seems to be related to new captcha requests in the eufy app, if I had to guess

**Comment by schliemann** [comment]
I think this is an issue with the client library that is not yet fixed and is caused by Eufy.

bropat/eufy-security-client#242

**Comment by schliemann** [comment]
Beta has been updated with @baeuric's PR.

**Comment by andypburke** [comment]
Thanks @schliemann i have installed beta version and the camera works although an error is still thrown up in the logs 

![1D0DF7E3-D6EE-47B4-80C5-E2675256051D](https://user-images.githubusercontent.com/4771045/201474299-bad4a0fa-a76b-4311-9b1e-1d33f389f9fc.jpeg)

**Comment by Lokihb34** [comment]
Update: it works with one minor exception: debug shows error "handleSecuritySystemCurrentStateGet Wrong return value" every now and then. My setup is Homebridge version 1.5.1, Homebridge UI 4.50.2, and the BETA Eufy plugin 2.2.0 - beta.6. I think it's running node.js v 16.x.x. Hope this helps others w/ issue

**Comment by schliemann** [comment]
> Update: it works with one minor exception: debug shows error "handleSecuritySystemCurrentStateGet Wrong return value" every now and then. My setup is Homebridge version 1.5.1, Homebridge UI 4.50.2, and the BETA Eufy plugin 2.2.0 - beta.6. I think it's running node.js v 16.x.x. Hope this helps others w/ issue

You should probably make a new issue on this as I don't think it is related.

**Comment by schliemann** [comment]
> Thanks @schliemann i have installed beta version and the camera works although an error is still thrown up in the logs
> 
> ![1D0DF7E3-D6EE-47B4-80C5-E2675256051D](https://user-images.githubusercontent.com/4771045/201474299-bad4a0fa-a76b-4311-9b1e-1d33f389f9fc.jpeg)

Yeah. This error is a bit weird since setStatusLed certainly should be available on stations.

**Comment by schliemann** [comment]
> Thanks @schliemann i have installed beta version and the camera works although an error is still thrown up in the logs
> 
> ![1D0DF7E3-D6EE-47B4-80C5-E2675256051D](https://user-images.githubusercontent.com/4771045/201474299-bad4a0fa-a76b-4311-9b1e-1d33f389f9fc.jpeg)

Ok. Found the issue we are seeing. It is caused by version 2.2.3:

https://github.com/bropat/eufy-security-client/commit/5cdcf128c09c26417894019137b5364af3809565

Will try and fix these. Will move them to my RPi tonight.

**Comment by schliemann** [comment]
@andypburke Please try 2.2.0-beta.7 to see if the errors regarding setStatusLed is solved.

**Comment by Mech4711** [comment]
No Errors for me at lease anymore

**Comment by csoprano** [comment]
Hi,
 the problem is partially solved. But now I can't use the switches to enable/disable the camera and motion sensor.
In the logs I get this error:
[15/11/2022, 09:41:26] [homebridge-eufy-security] This plugin threw an error from the characteristic 'On': Unhandled error thrown inside write handler for characteristic: station.setMotionDetection is not a function. See https://homebridge.io/w/JtMGR for more info.

**Comment by schliemann** [comment]
> Hi, the problem is partially solved. But now I can't use the switches to enable/disable the camera and motion sensor. In the logs I get this error: [15/11/2022, 09:41:26] [homebridge-eufy-security] This plugin threw an error from the characteristic 'On': Unhandled error thrown inside write handler for characteristic: station.setMotionDetection is not a function. See https://homebridge.io/w/JtMGR for more info.

Please create a seperate issue.

**Comment by csoprano** [comment]
I correct myself. with beta 7 everything works.
thank you

**Comment by schliemann** [comment]
Non-beta is updated. If someone could please test that login works again.

**Comment by yanavery** [comment]
With release 2.1.3, I still can't login ... Homebridge logs say:
```
[12/18/2022, 6:47:15 PM] [EufySecurity-2.1.3] ERROR: Not connected can't continue!
```
Different login attempts with either my main Eufy account or the one I created for Homebridge purposes are both failing while those 2 accounts are working fine when used from the Eufy app. I'm also seeing this in the Homebridge logs after I attempt a login:
```
[18/12/2022, 19:23:54] [Homebridge UI] [homebridge-eufy-security] Incoming Request: /login
[18/12/2022, 19:27:13] [Homebridge UI] [homebridge-eufy-security] Terminating child process...
[18/12/2022, 19:27:13] [Homebridge UI] [homebridge-eufy-security] Child process ended
```
When I'm trying to use my main account which has 2FA configured, I do receive the Eufy email and I do get challenged to enter my 2FA code in the Homebridge Eufy Plug-in, but it fails after ... For my other Eufy account that doesn't have 2FA enabled, I obviously don't get challenged for a 2FA code and fails immediately with a message that says `"Login was not successful!"` ... I do get some Captha challenges along the way, so looks like some communication between Homebridge and Eufy is happening, it just doesn't seem to be able to finalize.

BTW, Eufy country for me is set to `Canada`, maybe it could be related to that?

Thanks for any help ...

**Comment by jp-lno** [comment]
Same behavior for me. Both release and beta version. Has worked for a while and now not possible to connect on start

**Comment by schliemann** [comment]
@yanavery You shouldn't comment on an old closed issue.

Please create a new one for better tracking.
